src/NLog/LayoutRenderers/Wrappers/WrapperLayoutRendererBase.cs

```diff
@@ -49,14 +51,32 @@ namespace NLog.LayoutRenderers.Wrappers
     /// </example>
     public abstract class WrapperLayoutRendererBase : LayoutRenderer
     {
+        private Layout _inner;
+
         /// <summary>
         /// Gets or sets the wrapped layout.
         /// 
         /// [DefaultParameter] so Inner: is not required if it's the first
         /// </summary>
         /// <docgen category='Transformation Options' order='10' />
         [DefaultParameter]
-        public Layout Inner { get; set; }
+        public Layout Inner
+        {
+            get => _inner;
+            set
+            {
+                var changed = !ReferenceEquals(_inner, value);
+                _inner = value;
+
+                if (changed)
+                    InnerChanged?.Invoke(this, EventArgs.Empty);
+            }
+        }
+
+        /// <summary>
+        /// Notify when <see cref="Inner"/> has been changed
+        /// </summary>
+        protected event EventHandler InnerChanged;
```

Comment:
Could InnerChanged be made internal until needed by others? I was actually thinking that this ReferenceEquals code could just be in ObjectPathRendererWrapper.RenderInnerAndTransform (ReferenceEquals between Inner and _innerAsSimple). No need to have the caching-optimization for one renderer bleeding into the base-class. Not like the logic about caching disappears from ObjectPathRendererWrapper

Reply:
Unfortunately c# 7.3 is still not working (private protected :( )

Reply:
done!